qt-bugs@ issue : 227003
Trolltech task ID : 227065
applied: no
author: Christian Ehrlicher <ch.ehrlicher@gmx.de>

add -(no)make configure options to configure.exe


Index: tools/configure/configureapp.cpp
===================================================================
--- tools/configure/configureapp.cpp	(revision 859598)
+++ tools/configure/configureapp.cpp	(working copy)
@@ -213,6 +213,7 @@
     dictionary[ "QT_SOURCE_TREE" ]    = fixSeparators(sourcePath);
     dictionary[ "QT_BUILD_TREE" ]     = fixSeparators(buildPath);
     dictionary[ "QT_INSTALL_PREFIX" ] = fixSeparators(installPath);
+    buildParts = QString( "libs tools examples demos docs translations" ).split(' ');
 
     dictionary[ "QMAKESPEC" ] = getenv("QMAKESPEC");
     if (dictionary[ "QMAKESPEC" ].size() == 0) {
@@ -457,6 +458,29 @@
             dictionary[ "SHARED" ] = "yes";
         else if( configCmdLine.at(i) == "-static" )
             dictionary[ "SHARED" ] = "no";
+        else if( configCmdLine.at(i) == "-nomake" ) {
+            ++i;
+            if (i==argCount)
+                break;
+            const QString part = configCmdLine.at(i);
+            if (part == "libs")
+                cout << "WARNING: buildpart " << part << " is needed!" << endl;
+            else
+                buildParts.removeAll(part);
+        }
+        else if( configCmdLine.at(i) == "-make" ) {
+            ++i;
+            if (i==argCount)
+                break;
+            const QString part = configCmdLine.at(i);
+            static const QStringList sl = QString( "libs tools examples demos docs translations" ).split(' ');
+            if (!sl.contains(part))
+                cout << "WARNING: Unknown buildpart " << part << endl;
+            else {
+                if(!buildParts.contains(part))
+                    buildParts.append(part);
+            }
+        }
 
 #endif
 
@@ -2342,7 +2366,7 @@
             cacheStream << (*var) << endl;
         }
         cacheStream << "CONFIG         += " << qmakeConfig.join( " " ) << " incremental create_prl link_prl depend_includepath QTDIR_build" << endl;
-        cacheStream << "QT_BUILD_PARTS  = libs tools examples demos docs translations" << endl;
+        cacheStream << "QT_BUILD_PARTS  = " << buildParts.join( " " ) << endl;
         QString targetSpec = dictionary.contains("XQMAKESPEC") ? dictionary[ "XQMAKESPEC" ] : dictionary[ "QMAKESPEC" ];
         QString mkspec_path = fixSeparators(sourcePath + "/mkspecs/" + targetSpec);
         if(QFile::exists(mkspec_path))
Index: tools/configure/configureapp.h
===================================================================
--- tools/configure/configureapp.h	(revision 859598)
+++ tools/configure/configureapp.h	(working copy)
@@ -112,6 +112,7 @@
     QStringList disabledModules;
     QStringList enabledModules;
     QStringList modules;
+    QStringList buildParts;
 //    QStringList sqlDrivers;
     QStringList configCmdLine;
     QStringList qmakeConfig;
